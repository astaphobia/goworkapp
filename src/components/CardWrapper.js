import React from 'react'
import PropTypes from 'prop-types'

class Card extends React.Component {
    static propTypes = {
        item: PropTypes.object,
        onClick: PropTypes.func
    }
    render() {
        const { item, onClick } = this.props
        return (
            <div className="card job-card" style={{ margin: '0 0 20px 0', cursor: 'pointer' }} onClick={onClick}>
                <div className="card-body">
                    {item && (
                        <React.Fragment>
                            <div className="card-header">
                                <h5 className="card-title">
                                    {item.title}
                                </h5>
                                <h6 className="card-subtitle mb-2 text-muted">
                                    {item.company}
                                </h6>
                            </div>
                            <h6 className="card-subtitle mb-2 text-muted">
                                <i className="fa fa-map-marker" /> {item.location} {' '}
                                <i className="fa fa-briefcase" /> {item.type}
                            </h6>
                            <p className="card-text">
                                {Object.keys(item.requirements).reduce((str, key) => str += ` ${key}: ${item.requirements[key].join(', ')}`, '')}
                            </p>
                        </React.Fragment>
                    ) || 'Uppss zonk'}
                </div>
                <div className="card-footer text-muted">
                    <div className="row">
                        <div className="col-4">
                            <button className="btn btn-white">
                                <i className="fa fa-share-alt" /> Kongsi
                            </button>
                        </div>
                        <div className="col-4">
                            <button className="btn btn-white">
                                <i className="fa fa-heart" /> Simpan
                            </button>
                        </div>
                        <div className="col-4">
                            <button className="btn btn-white">
                                <i className="fa fa-pencil-square-o" /> Mohon
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
class CardWrapper extends React.Component {
    static propTypes = {
        children: PropTypes.array
    }
    static Card = Card
    render() {
        return (
            <div className="list">
                {this.props.children.length >= 1 && (
                    <React.Fragment>
                        {this.props.children}
                    </React.Fragment>
                ) || (
                        <Card />
                    )}
            </div>
        )
    }
}

export default CardWrapper