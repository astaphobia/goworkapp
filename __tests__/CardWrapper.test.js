import React from 'react'
import { shallow } from 'enzyme'

import CardWrapper from '@/components/CardWrapper'

describe('Wrapper', () => {
    const items = [
        { name: 'mock' },
        { name: 'mock-2' }
    ]
    it('hope doing well wrapper', () => {
        const comp = shallow(
            <CardWrapper>
                {items.map((item, index) => <CardWrapper.Card item={item} key={index} />)}
            </CardWrapper>
        )
        expect(comp).toMatchSnapshot()
    })
})