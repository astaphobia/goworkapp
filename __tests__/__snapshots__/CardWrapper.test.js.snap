// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper hope doing well wrapper 1`] = `
<div
  className="list"
>
  <Card
    item={
      Object {
        "name": "mock",
      }
    }
    key="0"
  />
  <Card
    item={
      Object {
        "name": "mock-2",
      }
    }
    key="1"
  />
</div>
`;
